src/expac.c

```diff
@@ -208,6 +212,16 @@ static int parse_options(int *argc, char **argv[])
         opt_listdelim = optarg;
         break;
       case 'H':
+        if (!optarg) {
+          /* getopt_long(3) is a little quick to decide no argument. If the
```

Comment:
There's no ambiguity. It sounds like you're suggesting that you want `-H G` to work, but that isn't the spec. `getopt_long(3)` clearly states:

```
Two colons mean an option takes an optional arg; if there is text in the current argv-element (i.e.,
in the same word as the option name itself, for example, "-oarg"), then it is returned in optarg,
otherwise optarg is set to zero.
```

What you're doing here is actually providing backwards compatibility such that `-H G` *does* work, but you're breaking otherwise reasonable assumptions about how optional arguments operate across all callers of GNU's getopt_long.

The middle ground here is to not change the argument arity but instead declare a sentinel value of 'auto' which does what you're proposing. That keeps backwards compatibility but also allows (effectively) non-deterministic units.

Reply:
Yes, I was trying to be cute. Ended up just writing my own option-parsing library.

>declare a sentinel value of 'auto'

implemented this